### PR TITLE
fix: RFDETR OpenVINO export issues

### LINKED ISSUE #5882: RFDETR OpenVINO produces 0 despite pre-processing
**Describe the bug**

Due to known bug the new RFDETR model after export produces 0 accuracy. 
We need to provide a workaround before the release

**Steps to Reproduce**

1.
2.
3.
4.

**Environment:**

- OS: 
- Framework version: 
- Python version:
- OpenVINO version: 2026.0.0
- CUDA/cuDNN version:
- GPU model and memory:

### PR DESCRIPTION
**Pull request by omkar-334**
### Summary
  1. Box format - cxcywh → xyxy conversion in export                                                                                                
  2. f32 precision - prevent LayerNorm error compounding across 12 DINOv2 layers                                                                    
  3. Confidence threshold - disable pre-filtering so metric sees all detections 
  4. Box rescaling - align prediction and GT coordinate systems   
                                                                                                                                                 
May resolve #5882

### Checklist

- [x] The PR title and description are clear and descriptive
- [x] I have manually tested the changes
- [x] All changes are covered by automated tests
- [x] All related issues are linked to this PR (if applicable)

### CODE REVIEW

File: library/src/otx/backend/native/models/detection/detectors/rfdetr.py [hunk 1]
```diff
@@ -181,6 +182,10 @@ def export(
             index=box_index.unsqueeze(-1).repeat(1, 1, pred_boxes.shape[-1]),
         )
 
+        # RFDETR's pred_boxes are in normalized cxcywh format, but the
+        # OpenVINO SSD model type expects xyxy coordinates (matching RTDETR).
+        boxes = box_convert(boxes, in_fmt="cxcywh", out_fmt="xyxy")
+
```
[the remark this answers is not in the document]
Comment (omkar-334) [reply library/src/otx/backend/native/models/detection/detectors/rfdetr.py:188]: resolved

**Comment by kprokofi** [comment]
Hi @omkar-334 
Thank you for your contribution.
Did you check the mAP accuracy after export?

**Comment by omkar-334** [comment]
hey @kprokofi  
Screenshot 2026-03-31 at 8 34 42 AM  
The flow for testing this was - 
1. load model and train
2. test pytorch model
3. export to openvino
4. test openvino model
5. compare

**Comment by omkar-334** [comment]
hey @kprokofi 
1. The first fix is - RFDETR's decoder outputs boxes in cxcywh format but the exported model uses ModelAPI's SSD parser which reads boxes as xyxy. I thought this was the only fix but i still faced the 0 mAP error.  

2. I went through each layer of the backbone and found out that - OpenVINO defaults to fp16 internal computation on some devices. RFDETR's DINOv2 backbone has 12 transformer layers with LayerNorm - the fp16 rounding error compounds across layers. 
So i added `INFERENCE_PRECISION_HINT=f32` and now all the detections were matching. But mAP was still 0 . 

3. I Checked what ModelAPI was actually returning -> only 1 detection per image. The exported IR had confidence_threshold=0.275 in metadata. The trained model (30 epochs on a tiny dataset) had max scores around 0.08. ModelAPI was filtering out every detection before the metric could see them. PyTorch's evaluation path doesn't pre-filter -> it passes all 300 detections to the metric which applies its own thresholding.   
So I set `confidence_threshold=0.0` in ModelAPI configuration. The detection count went from 1 to a lot more. 

4. The mAP increased but the difference was still a lot. I inspected the coordinates. ModelAPI returned boxes at 512×512 scale (the model's input resolution), but ground truth boxes were at 64×64 scale (the original image size). The metric compared them directly via IoU. This is wrong.  
I added rescaling to convert predicted boxes from `img_shape` to `ori_shape` coordinates.   

With this, the mAP comparison is   
image

**Comment by kprokofi** [comment]
Thank you for the PR. It was included in different one along with fixing other issues.

**Comment by omkar-334** [comment]
im glad it came of use!
